[move] Remove `spec` syntax and ignore new-style specs

cos opened the pull request:
## Description 

 - Remove the deprecated `spec` and `invariant` keywords, and associated data structures.
 - Add functionality to ignore new-style specs. Similar to `test_only`, but more extensive, covering system code as well.

See [this test](https://github.com/MystenLabs/sui/pull/20687/files#diff-b0f3a7b1a4dbe3c3949399b0ef80dc9a7be5beefaea9c00edbb73335cb19cd42) and other files in the same directory for examples of filtered out code.

## Test plan 

 - Tests for the removed specs. They now throw "unrecognized syntax" errors instead of deprecation. 
 - Tests for the filtering; code still compiling even when mentioning specs.

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:
- [ ] REST API:

cos commented:
@tzakian, I see you are a reviewer for #20339, any chance we could get CI to run on this PR? It contains a subset of the changes.

cos commented:
Thanks! Could we get another run?

dariorussi commented:
@cos do I understand it correctly that this is only the change that you would like to upstream to the main repo to push your code to the move nursery?

cos commented:
@dariorussi yes, these are the only changes -- we might follow up with a minor PR in case we missed something.

Just pushed again, fixed a bug and renamed "verify" to "spec" to reduce any confusion with signature verification.

Emadalshamery commented:
Yes

cos commented:
We'll split this into multiple PRs to help with merging. First PR, with only spec syntax removal: https://github.com/MystenLabs/sui/pull/21684
